peerconnection_go_test.go

```diff
@@ -658,14 +658,23 @@ func TestOnICEGatheringStateChange(t *testing.T) {
 	peerConn, err := NewAPI(WithSettingEngine(s)).NewPeerConnection(Configuration{})
 	assert.NoError(t, err)
 
-	peerConn.OnICEGatheringStateChange(func(s ICEGathererState) {
+	var onStateChange func(s ICEGathererState)
+	onStateChange = func(s ICEGathererState) {
+		// Access to ICEGatherer in the callback must not cause dead lock.
+		peerConn.OnICEGatheringStateChange(onStateChange)
+		if state := peerConn.iceGatherer.State(); state != s {
+			t.Errorf("State change callback argument (%s) and State() (%s) result differs",
+				s, state,
+			)
+		}
+
 		switch s {
 		case ICEGathererStateClosed:
 			close(seenClosed)
 			return
 		case ICEGathererStateGathering:
 			if seenComplete.get() {
-				t.Fatal("Completed before gathering")
+				t.Error("Completed before gathering")
```

Comment:
Why change?

Reply:
Fatal (Fail) is only suitable from outer Test* function. It can't escape from `TestOnICEGatheringStateChange` in this case.

Reply:
https://golang.org/pkg/testing/#T.FailNow
> FailNow must be called from the goroutine running the test or benchmark function, not from other goroutines created during the test. Calling FailNow does not stop those other goroutines.

Reply:
:+1: thanks good to know :)